### ISSUE TITLE
BalanceTransactionService.ListAutoPaging Null Reference Exception

### ISSUE THREAD

**Issue by davetheunissen** [stripe/stripe-dotnet#1717]
When using the `ListAutoPaging` feature for `BalanceTransactions`, the service throws a null reference exception as it tries to go past the number of records in the result set. 

i.e. the default option is to limit the returned result set to 10 records. 
if I run the following, 10 records will be added to my list and the loop will throw the exception when trying to access an 11th record. 

```csharp
foreach (var balanceTransaction in _balanceTransactionService.ListAutoPaging())
{
      balanceTransactions.Add(balanceTransaction);
}
```

OSVersion: Windows 10 : Build 17134
.NETVersion : dotnet core 2.1
StripeVersion: 27.16.0

**Comment by ob-stripe** [comment]
Hi @davetheunissen. Can you share the full stacktrace for the exception?

**Comment by davetheunissen** [comment]
Here you go... We are getting the same exception for charges as well. 

System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Stripe.net
  StackTrace:
   at Stripe.Service`1.<ListRequestAutoPaging>d__24`1.MoveNext()
   at Services.Stripe.StripeBalanceTransactionService.ListBalanceTransactions(BalanceTransactionListOptions balanceTransactionListOptions) in Services\Stripe\StripeBalanceService .cs:line 35
   at RIBS.StripeTestHarness.Controllers.BalanceController.List(String payoutId) in Controllers\BalanceController .cs:line 36
   at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeActionMethodAsync>d__12.MoveNext()

**Comment by ob-stripe** [comment]
Thanks! I think I know what the bug is. I'll update here once I confirm or if I need more information.

**Comment by ob-stripe** [comment]
Actually, can you try the following? It will help me confirm if the issue is what I think it is:

```c#
var options = new BalanceTransactionListOptions();
foreach (var balanceTransaction in _balanceTransactionService.ListAutoPaging(options))
{
      balanceTransactions.Add(balanceTransaction);
}
```

**Comment by davetheunissen** [comment]
Hi @ob-stripe, confirmed the issue is caused by a null options object even though the LastAutoPaging method has a default parameter for the options that accept nulls.

**Comment by ob-stripe** [comment]
Thanks @davetheunissen! We will release a fix soon.

**Comment by ob-stripe** [comment]
We just released version 27.16.1 which should fix the issue. Thanks for the report, and sorry for the inconvenience!

**Comment by davetheunissen** [comment]
Awesome thank you for the super quick response!
